### ISSUE TITLE
Support enabling secrets encryption after a cluster has been created

### ISSUE THREAD

**Issue by cjellick** [k3s-io/k3s#7195]
**Is your feature request related to a problem? Please describe.**
Secrets encryption isn't supported for existing clusters. Has to be turned on when the cluster is created. This is documented:
https://docs.k3s.io/cli/secrets-encrypt#encryption-key-rotation

> Starting K3s without encryption and enabling it at a later time is currently not supported.

So, this is just to ask for that to be supported (after talking with @brandond in rancher user slack).

**Describe the solution you'd like**
Don't have anything to add to this section.

**Describe alternatives you've considered**
One possibility could be to just to do a better job of not supporting it. We got our cluster pretty hosed by flipping this on.  Each server came up with a different encryption key, which caused madness trying to encrypt and decrypt secrets.

**Comment by cjellick** [comment]
oh, this is a little misleading other area of the docs: https://docs.k3s.io/security/secrets-encryption

implies you can do it

**Comment by brandond** [comment]
Ah indeed.  In once place we say
> Secrets-encryption cannot be enabled on an existing server without restarting it.
while another says
> Starting K3s without encryption and enabling it at a later time is currently not supported.

Functionally, single-servers cluster do currently work if you enable it later, but due to lack of proper support for updating the encryption config in the boostrap data, things will break if you try to do so on a multi-server cluster.

**Comment by riuvshyn** [comment]
Hello, just wanted to clarify enabling secrets-encryption on live k3s cluster is still not supported, right?

**Comment by brandond** [comment]
Still not supported.

**Comment by denniseffing** [comment]
Hi @brandond,
it has been over a year now since the last update, so I just wanted to ask: Are there any plans on implementing this feature request in the somewhat near future?

**Comment by dereknola** [comment]
No, there has been no movement it implement this feature, I've forgotten about it. I can try and allocate some time this month to look at it.

**Comment by klemmchr** [comment]
Hi @dereknola,
did you had any chance to look at this?

**Comment by dereknola** [comment]
EDIT: More work was required, see the actual PR below!

Original Post:
```
@klemmchr This was never intentionally developed into a feature, but looking at the existing code, this already works.

BUT you must rotate keys to ensure "past secrets" get encrypted. 

Full workflow:

1) You created a normal k3s cluster (no secrets encryption)
2) Time passes, secrets get added...
3) You restart the k3s cluster with `--secrets-encryption`.... A new encryption-config.json is created and kubeapiserver starts using it... BUT your old secrets are still not encrypted 
4) You run `k3s secrets-encrypt rotate-keys`, this creates a new encryption key and throws away the one you just started the server with... but it crucially reencrypts ALL secrets with new key, so unencrypted secrets from stage 1/2 are now encrypted. 

I need to validate this works on HA, but it currently works this way on single node.... also, I really need to start the process of deprecating/removing the old `k3s secrets-encrypt preprare/rotate/reencrypt` commands (don't use them they are slow and old).
```

**Comment by dereknola** [comment]
So this workflow doesn't quite work on HA and I don't see an easy way of enabling it nicely. The problem is that we use the config.yaml/flags as a source of truth. You need to have `secrets-encryption: true`, to generate the initial encryption-config. But we do this stage BEFORE we have synced the backend DB state of file, so what happens is that each server writes their own unique encryption-config.... I think Implementing `identity` as a provider would be the way to avoid this.

**Comment by klemmchr** [comment]
Would it be viable as a work around to copy the encryption config from the initial node to the other nodes? I tried it this way and it seemed to work fine, however I did not rotate keys beforehand so maybe it was just a fluke. Not saying that this is really user friendly but it could work for desperate people like me that just created a new cluster and missed to set secrets encryption from the beginning.

**Comment by aganesh-suse** [comment]
Closing based on: https://github.com/k3s-io/k3s/issues/13407

**Comment by denniseffing** [comment]
I'm just here to say thank you. So thank you very much for allocating your time on implementing this, I really appreciate it! Looking forward to rolling this out soon. 🥳
